.pre-commit-config.yaml

```diff
@@ -0,0 +1,51 @@
+default_language_version:
+  python: python3.6
+
+repos:
+  - repo: https://github.com/psf/black
+    rev: 20.8b1
+    hooks:
+      - id: black
+        name: Formatting Python
+
+  - repo: https://github.com/pycqa/pylint
+    rev: pylint-2.6.0
+    hooks:
+      - id: pylint
+        name: Lint Python (non-salt)
+        # Linting for non-salt files (also exclude tests files)
+        additional_dependencies: ['pyenchant~=2.0', 'doit~=0.32.0']
+        exclude: >
+          (?x)^(
+              salt/_.*|
+              salt/tests/.*|
+              tests/.*
+          )$
+      - id: pylint
+        name: Lint Salt
+        # Linting for salt files only in directory "salt/_<module>/<module_name>.py"
+        files: salt/_.*/.*\.py
+        additional_dependencies: ['saltpylint==2020.9.28']
+        args:
+          - --rcfile=salt/.pylintrc
+          - --ignore=metalk8s_package_manager_apt.py
+
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v0.740
+    hooks:
+      - id: mypy
+        name: Python typing
+        # Only check typing for typed files
+        files: >
+          (?x)^(
+              buildchain/dodo.py|
+              buildchain/buildchain/.*|
+              packages/debian/common/download_packages.py
+          )$
+        args:
+          - --strict
```

Comment:
Not super fond of this. Now this needs to be kept aligned between here and `tox.ini` (which should be the canonical source).

Reply:
I agree, for the moment that commands are really simple so... shouldn't be really hard to maintain but indeed I would prefer a single entry.
We can do something better if we change our tox so that we are able to run a command only for a specific file instead of always checking/acting on all files

Reply:
Any opinion on this one @scality/metalk8s ?

Reply:
Couldn't we call `tox` from a pre-commit hook? And we could also render the hook to the right place using `doit`, no need for another tool to `pre-commit install` (personally just discovered it :sweat_smile:).

Reply:
> we need to rewrite a bit our tox so that we are able to run only on one specific file.

Likely a good idea, yes.

> WDYM ?

That instead of using this `pre-commit` tool to install a magic `.git/hooks/pre-commit` script which parses this `.pre-commit-config.yaml`, just put our own `.git/hooks/pre-commit` script and have it do whatever we need. Unless you really think we should adopt this tool as a standard that we expect developers to have pre-installed (otherwise they can't really enable this hook).

Reply:
For me `pre-commit` is kind of common in the Python world and it's not because we have a pre-commit config, that all developer needs to have pre-commit hook installed it's just to help the developer if they want to.
Also, a dev can just run `tox -e pre-commit` to run this pre-commit "hook" manually and do not need to install pre-commit directly. Or just wait for a build that will basically check that linting&co is good as well.

IMHO we should define the lining stuff directly in pre-commit config and in tox we just call pre-commit on all files, from what I saw on various python project it's something fairly common